File: Leetcode-one-by-one/src/Can Convert String in K Moves.go
func canConvertString(s string, t string, k int) bool {
	if len(s) != len(t) {
		return false
	}

	cnt := make([]int, 26)

	for i := 0; i < len(s); i++ {
		diff := (int(t[i])-int(s[i]) + 26) % 26

		if diff == 0 {
			continue
		}

		requiredMove := diff + 26*cnt[diff]

		if requiredMove > k {
			return false
		}

		cnt[diff]++
	}

	return true
}

File: Leetcode-one-by-one/src/Can Convert String in K Moves.java
class Solution {
    public boolean canConvertString(String s, String t, int k) {
        if (s.length() != t.length()) {
            return false;
        }

        int[] cnt = new int[26];

        for (int i = 0; i < s.length(); i++) {
            int diff = (t.charAt(i) - s.charAt(i) + 26) % 26;

            if (diff == 0) {
                continue;
            }

            int requiredMove = diff + 26 * cnt[diff];

            if (requiredMove > k) {
                return false;
            }

            cnt[diff]++;
        }

        return true;
    }
}
